test/e2e/specs/site-editor/template-revert.spec.js

```diff
@@ -0,0 +1,307 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	templateRevertUtils: async ( { page }, use ) => {
+		await use( new TemplateRevertUtils( { page } ) );
+	},
+} );
+
+test.describe( 'Template Revert', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+	test.beforeEach( async ( { admin, requestUtils } ) => {
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await admin.visitSiteEditor();
+	} );
+
+	test( 'should delete the template after saving the reverted template', async ( {
+		editor,
+		page,
+		templateRevertUtils,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Test' },
+		} );
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		await page.click( 'role=button[name="Show template details"i]' );
+
+		// The revert button isn't visible anymore.
+		await expect(
+			page.locator( 'role=menuitem[name=/Clear customizations/i]' )
+		).not.toBeVisible();
+	} );
+
+	test( 'should show the original content after revert', async ( {
+		editor,
+		templateRevertUtils,
+	} ) => {
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Test' },
+		} );
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		const contentAfter = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfter );
```

Comment:
Nit: Since it's `expect( received ).toBe( expected )`, we should probably switch them? It'll affect the error message when it fails.

```suggestion
		expect( contentAfter ).toEqual( contentBefore );
```

Ditto to the same things below.

Reply:
Oh, I kept the order from the original test 😅  but yes, it does make more sense the other way around.